File: 824-number-of-lines-to-write-string/number-of-lines-to-write-string.go
func numberOfLines(widths []int, s string) []int {
    width, line := 0, 1
    for i := range s {
        width += widths[s[i] - 'a']
        if width > 100 {
            line++
            width = widths[s[i] - 'a']
        }
    }

    return []int{line, width}
}

File: 824-number-of-lines-to-write-string/number-of-lines-to-write-string.py
class Solution:
    def numberOfLines(self, widths: List[int], s: str) -> List[int]:
        lines = 1
        current_width = 0
        
        for char in s:
            char_idx = ord(char) - ord('a')
            char_w = widths[char_idx]
            
            if current_width + char_w > 100:
                lines += 1
                current_width = char_w
            else:
                current_width += char_w
                
        return [lines, current_width]